bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -0,0 +1,120 @@
+args = parser.parse_args()
+init_logging(args.verbose)
+
+def kde_awkde(x, x_grid, alp=0.5, gl_bandwidth='silverman', ret_kde=False):
+    kde = akde.GaussianKDE(glob_bw=gl_bandwidth, alpha=alp, diag_cov=True)
+    kde.fit(x)
+    if isinstance(x_grid, (list, tuple, numpy.ndarray)) == False:
+        y = kde.predict(x_grid)
+    else:
+        y = kde.predict(x_grid)
+    if ret_kde == True:
+        return kde, y
+    return y
+
+def kfcv_awkde(sample, bwchoice, alphachoice, k=2):
+    if bwchoice not in ['silverman', 'scott']:
+        bwchoice = float(bwchoice)
+    fomlist = []
+    kfold = kf.KFold(n_splits=k, shuffle=True, random_state=None )
+    for train_index, test_index in kfold.split(sample):
+        train, test = sample[train_index], sample[test_index]
+        y = kde_awkde(train, test, alp=alphachoice, gl_bandwidth=bwchoice)
+        fomlist.append(numpy.sum(numpy.log(y)))
+    return numpy.mean(fomlist)
+
+def optimizedparam(sampleval, nfold=2):
+    bwgrid = ['scott', 'silverman']+numpy.logspace(-2,0,10).tolist()
+    alphagrid = [1]
+    FOM= {}
+    for gbw in bwgrid:
+        for alphavals in alphagrid:
+            FOM[(gbw, alphavals)] = kfcv_awkde(sampleval, gbw, alphavals, k=nfold)
+    optval = max(FOM.items(), key=operator.itemgetter(1))[0]
+    optbw, optalpha  = optval[0], optval[1]
+    maxFOM = FOM[(optbw, optalpha)]
+    return optbw, optalpha
+
+#calling template data and constructing template_kde_hdf file over template params
+template_file = h5py.File(args.template_file, 'r')
+mass1_n = template_file['mass1'][:]
+mass2_n = template_file['mass2'][:]
+spin1z_n = template_file['spin1z'][:]
+spin2z_n = template_file['spin2z'][:]
+chi_eff_n = convert.chi_eff(mass1_n, mass2_n, spin1z_n, spin2z_n)
+mchirp_n = convert.mchirp_from_mass1_mass2(mass1_n, mass2_n)
+eta_n = convert.eta_from_mass1_mass2(mass1_n, mass2_n)
+
+if args.make_template_kde:
+    sample_n = numpy.vstack((numpy.log(mchirp_n), eta_n, chi_eff_n)).T
+    grid_pts_n = sample_n
+    optbw_n, optalpha_n = optimizedparam(sample_n, nfold=2)
+    kde_data_n = kde_awkde(sample_n, grid_pts_n, alp=optalpha_n, gl_bandwidth=optbw_n)
+
+    hdf = h5py.File(args.output_file, 'w')
+    hdf.create_dataset('kde_template', data=kde_data_n)
+    hdf.attrs['bank_kde'] = 'kde_template'
+    hdf.create_dataset('mass1', data=mass1_n)
+    hdf.create_dataset('mass2', data=mass2_n)
+    hdf.create_dataset('spin1z', data=spin1z_n)
+    hdf.create_dataset('spin2z', data=spin2z_n)
+    hdf.create_dataset('mchirp', data=mchirp_n)
+    hdf.create_dataset('eta', data=eta_n)
+    hdf.create_dataset('chi_eff', data=chi_eff_n)
+    hdf.create_dataset('grid_pts', data=grid_pts_n)
+    hdf.close()
+
+#calling signal data and constructing signal_kde_hdf file over template params
+if args.make_signal_kde:
+    data_signal = numpy.genfromtxt(args.signal_file, dtype = float, delimiter=',', names = True)
+    mass2 = data_signal['mass2']
+    mass2_threshold = 5
```

Comment:
This should be made into an input option, as in the future we may want to use a different limit or no limit. 
Since we are assuming the signal file has mass1 > mass2, the lower limit on mass2 is really a minimum component mass, so I would name the option something like '--min-mass'. 
Also in order to make sure that mass1 > mass2 for the signal points, you should put in a check here, eg `assert min(mass1 - mass2) > 0`
otherwise we might run into a case where some of the the signal KDE points have mass1<mass2.